src/cogs/interactions/slash/utils.py

```diff
@@ -0,0 +1,563 @@
+                "Accept-Encoding": "gzip, deflate, br, zstd",
+                "Accept": "*/*",
+                "Connection": "keep-alive",
+            }
+        ) as session:
+            async with session.get(
+                yarl.URL(
+                    "https://reddit.com/r/PESU/comments/14c1iym/.json", encoded=True
+                )
+            ) as response:
+                if response.status == 200:
+                    data = await response.json()
+                    x = data[0]["data"]["children"][0]["data"]["selftext"]
+                    finedata = {}
+                    y = x.split("# ")
+                    for i in y:
+                        j = i.split("\n\n")
+                        if "This post will be" in j[0]:
+                            continue
+                        # print(f"Processing Title: {j[0]}")
+                        s = j[1].split("* ")
+                        news = list(filter(None, s))
+                        for i in news:
+                            if ") or [" in i:
+                                chakdeh = i.split(") or [")
+                                for i in chakdeh:
+                                    l = i.split("](")
+                                    if l[0].startswith("["):
+                                        l[0] = l[0][1:]
+                                    if l[1].endswith(")"):
+                                        l[1] = l[1][:-1]
+                                    finedata.setdefault(j[0], []).append({l[0]: l[1]})
+
+                            else:
+
+                                chakdeh = i.split("](")
+                                if chakdeh[0].startswith("["):
+                                    chakdeh[0] = chakdeh[0][1:]
+                                if chakdeh[1].endswith(")"):
+                                    chakdeh[1] = chakdeh[1][:-1]
+
+                                if chakdeh[1].endswith("\n"):
+                                    chakdeh[1] = chakdeh[1][:-1]
+                                finedata.setdefault(j[0], []).append(
+                                    {chakdeh[0]: chakdeh[1]}
+                                )
+
+                    return finedata
+                else:
+                    resp = await response.text()
+                    print(
+                        f"Failed to fetch data: {response.status}, falling back to local data. {resp}"
+                    )
+                    with open("faq.json", "r") as file:
+                        data = json.load(file)
+                    return data
+
+    async def get_data(self):
+        if not self.cached_data:
+            self.cached_data = await self.fetch_data()
+        return self.cached_data
+
+    @app_commands.command(name="faq", description="Read the FAQ for PESU")
+    @app_commands.describe(
+        category="Optional category of the FAQ",
+        question="Optional specific question inside the category",
+    )
+    async def faq(
```

Comment:
Enhancement: Add an optional member parameter. If specified, the response message will mention that user